boomer/src/main.rs

```diff
@@ -43,14 +43,29 @@ struct CliArgs {
     verbose: u8,
 }
 
-fn find_connector(device: &Device, connector_name: Option<&str>) -> Option<Rc<Connector>> {
-    if let Some(name) = connector_name {
-        device.connectors().find(|con| con.to_string() == name)
+fn find_connector(device: &Device, connector_name: Option<&str>) -> Result<Rc<Connector>> {
+    let connectors = if let Some(name) = connector_name {
+        device
+            .connectors()
+            .filter(|con| con.to_string() == name)
+            .collect::<Vec<_>>()
     } else {
-        device.connectors().find(|con| {
-            con.connector_type() == ConnectorType::HDMIA
-                && con.status().unwrap_or(ConnectorStatus::Unknown) == ConnectorStatus::Connected
-        })
+        device
+            .connectors()
+            .filter(|con| {
+                con.connector_type() == ConnectorType::HDMIA
+                    && con.status().unwrap_or(ConnectorStatus::Unknown)
+                        == ConnectorStatus::Connected
+            })
+            .collect::<Vec<_>>()
```

Comment:
And this one can be something like:
```rust
let conn_iter = device
    .connectors()
    .filter(|con| { ... });

let first = conn_iter.next().ok_or(anyhow::anyhow!("No active connectors"))?;

if conn_iter.next().is_none() {
    Ok(first)
} else {
    Err(anyhow::anyhow!("Multiple active connectors, select one using the --connector-name argument"))
}
```

Reply:
Sorry for the delay looking into this review comments, fixed now